credentials/gcp.go

```
@@ -0,0 +1,171 @@
+		log.Println("Another common issue is if running in a headless environment, where sctl expects" +
+			" to be invoked with GOOGLE_APPLICATION_CREDENTIALS environment variable set.")
+		log.Fatalf("Failed definetly with error %v", err)
+	}
+	return json.Marshal(tok)
+}
+
+// ValidateContext scans the environment for a declaration of authentication via ENV
+// In the case of the GCP provider, we scan for GOOGLE_APPLICAITON_CREDENTIALS and return
+// an error if the environment variable is declared.
+func (gc GoogleCredential) ValidateContext() error {
+	_, isSet := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS")
+	if isSet {
+		return errors.New("Environment Variable GOOGLE_APPLICATION_CREDENTIALS is present, and will prevent sctl from using any user configured credentials")
+	}
+
+	return nil
+}
+
+func (gc GoogleCredential) formatCredential(token *oauth2.Token, rawClientData []byte) GoogleToken {
+	var clientJSON GoogleClientJSON
+	err := json.Unmarshal(rawClientData, &clientJSON)
+	if err != nil {
+		log.Fatalf("Error Unmarshalling client credentials: %v", err)
+	}
+	googleToken := GoogleToken{
+		ClientID:     clientJSON.ClientConfig.ClientID,
+		ClientSecret: clientJSON.ClientConfig.ClientSecret,
+		RefreshToken: token.RefreshToken,
+		TheType:      "authorized_user",
+	}
+
+	return googleToken
+}
+
+// Login initiates a CLI workflow to authenticate the user with offline credentials limited to
+// the KMS scope
+func (gc GoogleCredential) Login(c utils.Configuration, credentialName string) error {
+	err := gc.ValidateContext()
+	if err != nil {
+		log.Printf("Configuration issue detected. %v", err)
+	}
+	// Encode the Client Configuration json as a byte stream
+	clientConfig := []byte(c.GoogleClient.Data)
+	// Initialize the API client
+	config, err := google.ConfigFromJSON(clientConfig, "https://www.googleapis.com/auth/cloudkms")
+
+	if err != nil {
+		log.Fatalf("Unable to parse client config: %v", err)
```

should return error instead of log.fatal